### PR TITLE
Improve date handling for data pipeline

### LINKED ISSUE #68: Catch Nodata for S1 and DEM
See discussion in https://github.com/Clay-foundation/model/pull/54#pullrequestreview-1754125092

### PR DESCRIPTION
**Pull request by yellowcap**
If no match is found for a year, others are being tried until a match is found or all years have been tested

Closes #68

### CODE REVIEW

**Comment by yellowcap** [comment]
Also added tile size increase to 512x512 pixels to this PR, ref #78

File: scripts/datacube.py [hunk 1]
```diff
@@ -424,31 +427,49 @@ def convert_attrs_and_coords_objects_to_str(data):
...
-        subset = [int(dat) for dat in subset.split(",")]
-        print(f"Subsetting to {subset}")
-        pixels = [
-            part[:, subset[1] : subset[3], subset[0] : subset[2]] for part in pixels
-        ]
 
-    pixels = [part.compute() for part in pixels]
+    print(f"Starting algorithm for MGRS tile {tile['name']} with index {index}")
+
+    # Shuffle years, use index as seed for reproducability but no
```
Comment (weiji14) [inline scripts/datacube.py:437]:
```suggestion
    # Shuffle years, use index as seed for reproducibility but no
```

**Review by weiji14 — APPROVED** [review]
Just one typo, otherwise should be good.

**Comment by yellowcap** [comment]
Got this to work on batch and gave good results. Kicking off a new batch run as we speak, if all goes well we'll have 10x the data in a few hours 🤞🏽

**Comment by weiji14** [comment]
Good timing! I'm hoping to kick off a new training run with Soumya's code later, and can test things out on the new data batch.

**Pull request merged**